### ISSUE TITLE
socket.AF_UNIX is not supported

### ISSUE THREAD

**Issue by muromec** [kivy/python-for-android#163]
Hi!

I have tried to use local sockets in kivy app and noticed, that local sockets are disabled at compile time.

Python configure script fails to check "sys/un.h" header (which is sort of broken on android) and disables `HAVE_SYS_UN_H` flag:

```
    configure:6133: checking sys/un.h usability
    configure:6133: ccache arm-linux-androideabi-gcc -DANDROID -mandroid  -fomit-frame-pointer --sysroot /home/kivy/android/android-ndk-r8c/platforms/android-14/arch-arm -c -DANDROID -mandroid  -fomit-frame-pointer --sysroot /home/kivy/android/android-ndk-r8c/platforms/android-14/arch-arm  conftest.c >&5
```

In file included from /home/kivy/android/android-ndk-r8c/platforms/android-14/arch-arm/usr/include/sys/un.h:32,
                 from conftest.c:84:
        /home/kivy/android/android-ndk-r8c/platforms/android-14/arch-arm/usr/include/linux/un.h:18: error: expected specifier-qualifier-list before 'sa_family_t'

related android issue: https://code.google.com/p/android/issues/detail?id=53711

Possible solution is to either add workaround to configure script or force-enable `HAVE_SYS_UN_H` in `pyconfig.h`.
##

**Comment by joehunt588** [comment]
Hello muromec,how the progress? Did u manage solve it?i have counter same error ,i change it to AF_INET it working as charm,please reply if u have other methods,although it have been 3 years sorry for been late ^-^

**Comment by inclement** [comment]
Closing as probably fixed or different after all the many NDK changes over the years.

**Comment by Davincible** [comment]
I have tried this too, and while it did compile, I got an error while trying to bind to a local Unix socket address. Error: "OSError: [Errno 22] Invalid argument". 

For me it would be really usefull to get unix sockets to work to communicate with android services, since using AF_INET sockets works nice on first run, but if the app crashes and doesn't unbind properly the socket cannot be bound again until device restart because it's occupied.
